includes/core/class-mode.php

```diff
@@ -94,8 +73,11 @@ private function maybe_init() {
 		 */
 		$this->dev_mode = (bool) apply_filters( 'wcpay_dev_mode', $dev_mode );
 
-		$test_mode_setting = 'yes' === $this->gateway->get_option( 'test_mode' );
-		$test_mode         = $this->dev_mode || $test_mode_setting;
+		// Getting the gateway settings directly from the database so the gateway doesn't need to be initialized.
+		$settings_option_name = 'woocommerce_' . WC_Payment_Gateway_WCPay::GATEWAY_ID . '_settings';
+		$wcpay_settings       = get_option( $settings_option_name );
+		$test_mode_setting    = 'yes' === ( $wcpay_settings['test_mode'] ?? false );
+		$test_mode            = $this->dev_mode || $test_mode_setting;
```

Comment:
We are using this constant for consistency as it is the same constant WC uses for the gateway's `get_option` instance method.
I like having it in the Mode class as it is related to retrieving the mode setting. `WC_Payment_Gateway_WCPay` is already 4,5k lines long, so I think it is good to avoid adding more methods that don't need to be there.